ptf/sairif.py

```diff
@@ -8220,3 +8220,1408 @@ def mtuPacketStatsTest(self):
             'SAI_ROUTER_INTERFACE_STAT_IN_PACKETS'])
         self.assertTrue(self.vlan100_rif_counter_out == vlan100_rif_stats[
             'SAI_ROUTER_INTERFACE_STAT_OUT_PACKETS'])
+
+class RifToSubPortTest(PlatformSaiHelper):
+    """
+    """
+
+    def setUp(self):
+        super(RifToSubPortTest, self).setUp()
+
+        dmac0 = '00:11:22:33:44:55'
```

Comment:
All of the lines from here to runTest will need to be duplicated for all the other sub tests of L3SubPortTest which is where we picked this test. This is the reason we grouped them together.
But I think we can still split them into individual tests if we can sub-class better.

        self.pvMissTest()
        self.subPortToSubPortTest()
        self.subPortToRifTest()
        self.rifToSubPortTest()
        self.noFloodTest()
        self.vlanConflictTest()
        self.subPortAdminV4StatusTest()
        self.subPortAdminV6StatusTest()
        self.subPortV4MtuTest()
        self.subPortV6MtuTest()
        self.subPortIngressAclTest()
        self.subPortEgressAclTest()
        self.subPortECMPTest()
        self.subPortMyIPTest()
        self.subPortStatsTest()
        self.subPortNoTest()
        self.subPortQosGroupTest()

This is the list of tests for SubPorts grouped together today. They all have the same setup and teardown sequences. Only test in here which seems the odd one is the subPortQosGroupTest(). We can go about this 2 ways.
1. Just remove the subPortQosGroupTest() as a separate test and keep the rest together.
2. Create a new class called SubPortHelper and keep the common setUp and tearDown portions within there. This will avoid code duplication. We can then just keep doing new runTest classes for each sub test.

What do you say?

Reply:
@richardyu-ms

Reply:
I prefer the second approach, it is more like a complete solution.
We modified the case base on that, please help make a review.
thanks.